turbopy/diagnostics.py

```diff
@@ -314,6 +319,10 @@ def initialize(self):
 
         # Use composition to provide i/o functionality
         self.outputter = utilities[self._input_data["output_type"]](**self._input_data)
+        
+        # set up interval handler
+        if self.interval:
+            self.handler = IntervalHandler(self.interval, self.csv.write_data)
```

Comment:
Need to update this to use the `outputter` API, instead of hardcoded csv.
```suggestion
            self.handler = IntervalHandler(self.interval, self.outputter.write_data)
```